app/controllers/concerns/state_file/return_to_review_concern.rb

```diff
@@ -0,0 +1,22 @@
+module StateFile
+  module ReturnToReviewConcern
+    # This concern can be used by any controller that sometimes needs to redirect
+    # to the review page rather than the usual next page in the flow
+    extend ActiveSupport::Concern
+
+    private
+
+    def review_step
+      case params[:us_state]
+      when 'az'
+        StateFile::Questions::AzReviewController
+      when 'ny'
+        StateFile::Questions::NyReviewController
+      end
+    end
+
+    def next_step
+      params[:return_to_review].nil? ? super : review_step
+    end
```

Comment:
You could alternatively overwrite `next_path` and that might make for an simpler integration with questions controller. I'm not saying this as a suggestion so much as pointing out another option in case you'd like it better.

Reply:
i think i'm going to just push this as is, but it's not because i have a strong stylistic preference -- i was concerned that overriding all of next_path would always work properly because there's logic in there beyond selecting the next controller that wouldn't run. specifically this bit:

`        
if step_for_next_path.resource_name.present? && step_for_next_path.resource_name == self.class.resource_name
    options[:id] = current_resource.id
end
`

i'm not sure what that's for, but whatever it is, it seemed to make sense to be able to override just the next step, as opposed to all the logic around determining the next path.